### PR TITLE
Inline React elements in production

### PR DESCRIPTION
**Pull request by billyjanitsch**
See the [transform documentation](https://babeljs.io/docs/en/babel-plugin-transform-react-inline-elements).

This is a production-only optimization which speeds up the creation of React elements by avoiding a loop over props. The transform is smart enough to de-opt in problematic cases.

~I'd like to release this as `next` and [make sure it works](https://babeljs.io/docs/en/babel-plugin-transform-react-inline-elements#note) in one of our apps before tagging as `latest`. In theory, this should not be a breaking change.~ Out of caution, I'm going to release this as `next` in a major release.

### CODE REVIEW

**Comment by billyjanitsch** [comment]
To do: add a test to verify that this works with `{runtime: true}`.

**Comment by billyjanitsch** [comment]
With https://github.com/kensho/babel-preset-kensho/pull/59 landed, this needs docs.

**Comment by jhhayashi** [comment]
this sgtm. @nabeel- ?

File: test/snapshots/index.js.md [hunk 1]
```diff
@@ -55,12 +77,41 @@ Generated by [AVA](https://ava.li).
...
+      }, "abc");␊
+    }`
+
 ## removes prop types by default in production environment
 
 > Snapshot 1
 
-    `export default function MyComponent(props) {␊
-      return React.createElement("div", null);␊
+    `var REACT_ELEMENT_TYPE;␊
```
Comment (nabeel-) [inline test/snapshots/index.js.md]: Do we care that this test is modified as well? Doesn't seem like an issue but if we do then we should add `{inlineReactElements: false}` to the `options`

**Comment by billyjanitsch** [comment]
I don't think this transform will work well with the Emotion preset, and I've heard it suffers from some correctness issues as well, so I'm going to close this. Maybe we can revisit it someday.

**Pull request closed without merging**